evalml/data_checks/target_leakage_data_check.py

```diff
@@ -11,25 +13,49 @@
 
 
 class TargetLeakageDataCheck(DataCheck):
-    """Check if any of the features are highly correlated with the target."""
+    """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation."""
 
-    def __init__(self, pct_corr_threshold=0.95):
-        """Check if any of the features are highly correlated with the target.
+    def __init__(self, pct_corr_threshold=0.95, pearson_corr=False):
+        """Check if any of the features are highly correlated with the target by using mutual information or Pearson correlation.
 
-        Currently only supports binary and numeric targets and features.
+        If `pearson_corr=False`, this data check uses mutual information and supports all target and feature types.
+        Otherwise, it uses Pearson correlation and only supports binary with numeric and boolean dtypes.
+        Pearson correlation returns a value in [-1, 1], while mutual information returns a value in [0, 1].
 
         Arguments:
             pct_corr_threshold (float): The correlation threshold to be considered leakage. Defaults to 0.95.
+            pearson_corr (bool): Whether or not to use the Pearson correlation versus mutual information. Defaults to False, which uses mutual information.
```

Comment:
Not blocking and just a suggestion since this works for now, but I wonder if its better to have a set of strings determining the method. I can imagine if we add a third value, we'll have to change this to support that, too. But I guess this works for now 🤷

Reply:
@chukarsten recommended the same thing. Will change!